txstratum/prometheus.py

```diff
@@ -0,0 +1,90 @@
+import asyncio
+import os
+from typing import TYPE_CHECKING, Dict
+
+from prometheus_client import CollectorRegistry, Gauge, write_to_textfile  # type: ignore
+
+if TYPE_CHECKING:
+    from txstratum.manager import TxMiningManager
+
+
+# Define prometheus metrics and it's explanation
+METRIC_INFO = {
+    'miners_count': 'Number of connected miners',
+    'total_hashrate_ghs': 'Hashrate (Gh/s)',
+    'txs_solved': 'Number of solved transactions',
+    'txs_timeout': 'Number of timeouts when solving transactions',
+    'blocks_found': 'Number of blocks found',
+    'uptime': 'Service uptime',
+    'tx_queue': 'Number of transactions in the queue',
+    'block_template_error': 'Number of errors updating block template',
+}
+
+
+class PrometheusExporter:
+    """Class that sends hathor metrics to a node exporter that will be read by Prometheus."""
+
+    def __init__(self, manager: 'TxMiningManager', path: str, filename: str = 'tx-mining-service.prom'):
+        """Init PrometheusExporter.
+
+        :param metrics: Metric object that stores all the hathor metrics
```

Comment:
Wrong comment, this should be manager object. Maybe we could remove this, since we already have the type on the params.